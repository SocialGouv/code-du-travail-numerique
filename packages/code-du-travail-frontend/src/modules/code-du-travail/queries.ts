import { SOURCES } from "@socialgouv/cdtn-utils";
import { ElasticLaborCodeArticle } from "./type";
import { DocumentElasticResult, fetchDocument } from "../documents";

export const fetchLegalArticle = async (
  slug: string
): Promise<DocumentElasticResult<ElasticLaborCodeArticle> | undefined> => {
  return await fetchDocument<
    ElasticLaborCodeArticle,
    keyof DocumentElasticResult<ElasticLaborCodeArticle>
  >(["description", "title", "dateDebut", "html", "url", "notaHtml"], {
    query: {
      bool: {
        filter: [
          { term: { source: SOURCES.CDT } },
          { term: { slug } },
          { term: { isPublished: true } },
        ],
      },
    }
  });
};
