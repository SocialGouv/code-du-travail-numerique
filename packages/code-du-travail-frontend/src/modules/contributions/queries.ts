import * as Sentry from "@sentry/nextjs";
import { elasticDocumentsIndex, elasticsearchClient } from "../../api/utils";
import { SOURCES } from "@socialgouv/cdtn-utils";
import { DocumentElasticResult, fetchDocument, isSource } from "../documents";
import { Contribution, ContributionElasticDocument } from "./type";

export const fetchContributions = async <
  K extends keyof ContributionElasticDocument,
>(
  fields: K[],
  filters?: {
    cdtnIds?: string[];
  }
): Promise<Pick<ContributionElasticDocument, K>[]> => {
  const baseFilters: Array<any> = [
    { term: { source: SOURCES.CONTRIBUTIONS } },
    { term: { isPublished: true } },
  ];

  if (filters?.cdtnIds) {
    baseFilters.push({ terms: { cdtnId: filters.cdtnIds } });
  }

  const result = await elasticsearchClient.search<
    Pick<ContributionElasticDocument, K>
  >({
    query: {
      bool: {
        filter: baseFilters,
      },
    },
    size: 3000,
    _source: fields,
    index: elasticDocumentsIndex,
  });

  return result.hits.hits
    .map(({ _source }) => _source)
    .filter((source) => source !== undefined);
};

const formatContribution = (
  contribution: ContributionElasticDocument | undefined
): Contribution | undefined => {
  if (!contribution) {
    return undefined;
  }
  return {
    ...contribution,
    isGeneric: contribution.idcc === "0000",
    isNoCDT: contribution?.type === "generic-no-cdt",
    relatedItems: [
      {
        title: "Articles liÃ©s",
        items: contribution.linkedContent.reduce((arr, linked) => {
          if (!isSource(linked.source)) {
            Sentry.captureMessage(
              `la source de ${JSON.stringify(linked)} est incorrecte`
            );
            return arr;
          }
          return [
            ...arr,
            {
              title: linked.title,
              url: linked.slug,
              source: linked.source,
            },
          ];
        }, []),
      },
    ],
  };
};

export const fetchContributionBySlug = async (
  slug: string
): Promise<Contribution | undefined> => {
  const response = await fetchDocument<
    ContributionElasticDocument,
    keyof DocumentElasticResult<ContributionElasticDocument>
  >(
    [
      "metas",
      "idcc",
      "date",
      "title",
      "slug",
      "type",
      "linkedContent",
      "breadcrumbs",
      "ccSupported",
      "ccUnextended",
      "messageBlock",
      "references",
      "ccnShortTitle",
      "raw",
      "ficheSpDescription",
      "content",
      "url",
      "messageBlockGenericNoCDT",
    ],
    {
      index: elasticDocumentsIndex,
      query: {
        bool: {
          filter: [
            { term: { source: SOURCES.CONTRIBUTIONS } },
            { term: { slug } },
            { term: { isPublished: true } },
          ],
        },
      },
      size: 1,
    }
  );
  return formatContribution(response);
};
