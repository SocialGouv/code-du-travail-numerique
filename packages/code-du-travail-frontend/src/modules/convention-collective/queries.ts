import { ElasticAgreement } from "@socialgouv/cdtn-types";
import { orderByAlpha } from "../utils";
import { SOURCES } from "@socialgouv/cdtn-utils";
import { elasticDocumentsIndex, elasticsearchClient } from "../../api/utils";

type Props<K> = {
  fields?: K[];
  sortBy?: K;
  filterTitle?: string;
  size?: number;
};

export const fetchAllAgreements = async <K extends keyof ElasticAgreement>({
  fields,
  sortBy,
  filterTitle,
  size = 100,
}: Props<K>): Promise<Pick<ElasticAgreement, K>[]> => {
  const response = await elasticsearchClient.search<Pick<ElasticAgreement, K>>({
    query: {
      bool: {
        ...(filterTitle
          ? {
              should: [
                {
                  match: {
                    "shortTitle.french": {
                      boost: 0.9,
                      fuzziness: "1",
                      query: filterTitle,
                    },
                  },
                },
                {
                  match_phrase_prefix: {
                    "synonymes.french": {
                      query: filterTitle,
                    },
                  },
                },
                {
                  match_phrase_prefix: {
                    "title.french": {
                      query: filterTitle,
                    },
                  },
                },
              ],
            }
          : {}),
        filter: [
          { term: { source: SOURCES.CCN } },
          { term: { isPublished: true } },
          { term: { contributions: true } },
        ],
      },
    },
    size,
    ...(fields ? { _source: fields } : {}),
    index: elasticDocumentsIndex,
  });

  const data = response.hits.hits
    .map(({ _source }) => _source)
    .filter((item) => item !== undefined);
  if (sortBy) {
    data.sort((a, b) => orderByAlpha(a, b, sortBy));
  }
  return data;
};
