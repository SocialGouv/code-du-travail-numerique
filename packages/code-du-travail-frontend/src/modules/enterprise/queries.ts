import debounce from "debounce-promise";
import { SITE_URL } from "../../config";
import { Enterprise } from "./types";

export type SearchParams = {
  query: string;
  codesPostaux?: string[];
};

export const messageFetchSearchCcOrEnterprise =
  "Ce service est momentanément indisponible. Si vous utilisez un simulateur, vous pouvez tout de même poursuivre la simulation pour obtenir le résultat prévu par le Code du travail en sélectionnant l'option \"Je ne souhaite pas renseigner ma convention collective (je passe l'étape).\"";

const siretSirenError =
  "Veuillez indiquer un numéro Siret (14 chiffres) ou Siren (9 chiffres) valide";

const siretLengthError =
  "Veuillez indiquer un numéro Siret (14 chiffres obligatoire)";

const siretNumberError =
  "Veuillez indiquer un numéro Siret (14 chiffres uniquement)";

const apiEnterprises = function createFetcher(
  searchParams: SearchParams
): Promise<Enterprise[]> {
  if (
    /^\d{2,8}$/.test(searchParams.query.replace(/\s/g, "")) ||
    /^\d{4}[A-Za-z]$/.test(searchParams.query.replace(/\W/g, ""))
  ) {
    return Promise.reject(siretSirenError);
  }
  if (
    /^\d{10,13}$/.test(searchParams.query.replace(/\s/g, "")) ||
    /^\d{15,}$/.test(searchParams.query.replace(/\s/g, ""))
  ) {
    return Promise.reject(siretLengthError);
  }
  if (/\D+\d{14}/.test(searchParams.query.replace(/\s/g, ""))) {
    return Promise.reject(siretNumberError);
  }

  const url = `http://localhost:3000/api/enterprises?q=${encodeURIComponent(searchParams.query)}${
    searchParams.codesPostaux
      ? `&cp=${encodeURIComponent(searchParams.codesPostaux.join(","))}`
      : ""
  }`;

  return fetch(url)
    .then(async (response) => {
      if (response.ok) {
        const res = await response.json();
        return res.entreprises;
      }
      throw new Error();
    })
    .catch(() => {
      return Promise.reject(messageFetchSearchCcOrEnterprise);
    });
};

const searchEnterprises = debounce(apiEnterprises, 300);

export { searchEnterprises };
