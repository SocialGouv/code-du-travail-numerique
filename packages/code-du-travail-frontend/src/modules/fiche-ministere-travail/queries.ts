import { SOURCES } from "@socialgouv/cdtn-utils";

import { DocumentElasticResult, fetchDocument } from "../documents";
import {
  ElasticFicheTravailEmploi,
  ElasticFicheTravailEmploiSection,
} from "@socialgouv/cdtn-types";

export type FicheMT = DocumentElasticResult<ElasticFicheTravailEmploi> & {
  highlight?: ElasticFicheTravailEmploiSection;
  sections: ElasticFicheTravailEmploiSection[];
};

const formatFiche = (
  fiche: DocumentElasticResult<ElasticFicheTravailEmploi> | undefined
): FicheMT | undefined => {
  if (!fiche) {
    return undefined;
  }
  return {
    ...fiche,
    highlight: fiche.sections.find(
      (section) => !section.anchor || section.anchor === ""
    ),
    sections: fiche.sections.filter(
      (section) => section.anchor && section.anchor !== ""
    ),
  };
};

export const fetchFicheMT = async (
  slug: string
): Promise<FicheMT | undefined> => {
  return formatFiche(
    await fetchDocument<
      ElasticFicheTravailEmploi,
      keyof DocumentElasticResult<ElasticFicheTravailEmploi>
    >(
      [
        "breadcrumbs",
        "date",
        "description",
        "intro",
        "sections",
        "slug",
        "title",
        "url",
        "cdtnId",
      ],
      {
        query: {
          bool: {
            filter: [
              { term: { source: SOURCES.SHEET_MT_PAGE } },
              { term: { slug } },
              { term: { isPublished: true } },
            ],
          },
        },
        size: 1,
      }
    )
  );
};
