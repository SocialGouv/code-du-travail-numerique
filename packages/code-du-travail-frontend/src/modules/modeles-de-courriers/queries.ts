import {
  DocumentElasticWithSource,
  MailTemplateDoc,
} from "@socialgouv/cdtn-types";
import { elasticDocumentsIndex, elasticsearchClient } from "../../api/utils";
import { SOURCES } from "@socialgouv/cdtn-utils";
import { DocumentElasticResult, fetchDocument } from "../documents";

export const fetchAllModels = async <
  K extends keyof DocumentElasticWithSource<MailTemplateDoc>,
>(
  fields: K[]
): Promise<Pick<DocumentElasticWithSource<MailTemplateDoc>, K>[]> => {
  const response = await elasticsearchClient.search<
    DocumentElasticWithSource<
      Pick<DocumentElasticWithSource<MailTemplateDoc>, K>
    >
  >({
    query: {
      bool: {
        filter: [
          { term: { source: SOURCES.LETTERS } },
          { term: { isPublished: true } },
        ],
      },
    },
    size: 1000,
    _source: fields,
    index: elasticDocumentsIndex,
  });
  return response.hits.hits
    .map(({ _source }) => _source)
    .filter((model) => model !== undefined);
};

export const format = (model) => {
  if (model) {
    model.filesize = Math.round((model.filesize / 1000) * 100) / 100;
    if (model.filename.indexOf(".") > 0) {
      model.extension = model.filename.split(/\.([a-z]{2,4})$/)[1];
    }
  }

  return model;
};

export const fetchModel = async (filter: {
  slug?: string;
  _id?: string;
}): Promise<
  | (DocumentElasticResult<DocumentElasticWithSource<MailTemplateDoc>> & {
      extension: string;
    })
  | undefined
> => {
  const model = await fetchDocument<
    DocumentElasticWithSource<MailTemplateDoc>,
    keyof DocumentElasticResult<DocumentElasticWithSource<MailTemplateDoc>>
  >(
    [
      "breadcrumbs",
      "title",
      "meta_title",
      "date",
      "type",
      "html",
      "author",
      "filename",
      "filesize",
      "intro",
      "description",
      "metaDescription",
      "references",
    ],
    {
      query: {
        bool: {
          filter: [
            { term: { source: SOURCES.LETTERS } },
            { term: filter },
            { term: { isPublished: true } },
          ],
        },
      },
    }
  );

  return format(model);
};
