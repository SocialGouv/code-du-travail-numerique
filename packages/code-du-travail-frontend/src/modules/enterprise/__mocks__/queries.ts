import { SearchParams } from "../queries";
import { Enterprise } from "../types";

export const messageFetchSearchCcOrEnterprise =
  "Ce service est momentanément indisponible. Si vous utilisez un simulateur, vous pouvez tout de même poursuivre la simulation pour obtenir le résultat prévu par le Code du travail en sélectionnant l'option \"Je ne souhaite pas renseigner ma convention collective (je passe l'étape).\"";

export const searchEnterprises = (
  searchParams: SearchParams
): Promise<Enterprise[]> => {
  const query = searchParams.query.toLowerCase();

  if (query.includes("carrefour")) {
    return Promise.resolve([
      {
        activitePrincipale: "47.11F - Hypermarchés",
        etablissements: 208,
        highlightLabel: "CARREFOUR HYPERMARCHES",
        label: "CARREFOUR HYPERMARCHES",
        simpleLabel: "CARREFOUR HYPERMARCHES",
        matching: 208,
        siren: "451321335",
        address:
          "ZAE SAINT GUENAULT 1 RUE JEAN MERMOZ 91000 EVRY-COURCOURONNES",
        firstMatchingEtablissement: {
          siret: "45132133500023",
          address:
            "ZAE SAINT GUENAULT 1 RUE JEAN MERMOZ 91000 EVRY-COURCOURONNES",
        },
        complements: {
          liste_idcc: ["2216", "1486"],
        },
        conventions: [
          {
            id: "2216",
            contributions: true,
            num: 2216,
            shortTitle:
              "Commerce de détail et de gros à prédominance alimentaire",
            title:
              "Convention collective nationale du commerce de détail et de gros à prédominance alimentaire du 12 juillet 2001.  Etendue par arrêté du 26 juillet 2002 JORF 6 août 2002.",
            url: "https://www.legifrance.gouv.fr/affichIDCC.do?idConvention=KALICONT000005635085",
            slug: "2216-commerce-de-detail-et-de-gros-a-predominance-alimentaire",
          },
          {
            id: "1486",
            contributions: true,
            num: 1486,
            shortTitle:
              "Bureaux d'études techniques, cabinets d'ingénieurs-conseils et sociétés de conseils",
            title:
              "Bureaux d'études techniques, cabinets d'ingénieurs-conseils et sociétés de conseils",
            url: "https://www.legifrance.gouv.fr/affichIDCC.do?idConvention=KALICONT000005635173",
            slug: "1486-bureaux-detudes-techniques-cabinets-dingenieurs-conseils-et-societes-de",
          },
        ],
      },
    ]);
  }

  if (query.includes("bricomanie")) {
    return Promise.resolve([
      {
        activitePrincipale:
          "47.52Z - Commerce de détail de quincaillerie, peintures et verres en magasins spécialisés",
        etablissements: 85,
        highlightLabel: "BRICOMANIE",
        label: "BRICOMANIE",
        simpleLabel: "BRICOMANIE",
        matching: 85,
        siren: "123456789",
        address: "123 AVENUE DU BRICOLAGE 75000 PARIS",
        firstMatchingEtablissement: {
          siret: "12345678900123",
          address: "123 AVENUE DU BRICOLAGE 75000 PARIS",
        },
        complements: {
          liste_idcc: [],
        },
        conventions: [],
      },
    ]);
  }

  return Promise.resolve([]);
};
