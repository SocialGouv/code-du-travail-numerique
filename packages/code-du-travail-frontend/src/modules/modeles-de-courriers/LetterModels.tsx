"use client";

import { fr } from "@codegouvfr/react-dsfr";
import React, { useState } from "react";
import { ContainerList } from "../layout/ContainerList";
import { css } from "@styled-system/css";
import Card from "@codegouvfr/react-dsfr/Card";
import { getRouteBySource, SOURCES } from "@socialgouv/cdtn-utils";
import { summarize } from "../utils";
import { ModelDescriptionByRootTheme } from "./queries";
import Select from "@codegouvfr/react-dsfr/Select";

type Props = {
  modeles: ModelDescriptionByRootTheme;
};

export const LetterModels = ({ modeles }: Props) => {
  const [selectedTheme, setTheme] = useState<string>("");

  return (
    <ContainerList title="Modèles de documents" segments={[]}>
      <h1 id="modeles" className={fr.cx("fr-mt-0", "fr-mb-6w")}>
        Modèles de documents
      </h1>

      <p className={fr.cx("fr-text--md")}>
        Téléchargez et personnalisez les modèles de documents et de lettres pour
        vos démarches en lien avec le droit du travail
      </p>

      <Select
        className={`${fr.cx("fr-mt-6w")} ${list}`}
        label="Sélectionnez un thème"
        nativeSelectProps={{
          onChange: (event) => setTheme(event.target.value),
          value: selectedTheme,
        }}
      >
        <option value="">Tous les thèmes</option>
        {modeles.map(({ theme }) => (
          <option key={theme} value={theme}>
            {theme}
          </option>
        ))}
      </Select>

      {modeles
        .filter(
          (theme) => selectedTheme === "" || theme.theme === selectedTheme
        )
        .map(({ theme, modeles }) => {
          return (
            <>
              <h2 className={fr.cx("fr-mt-6w", "fr-h3")}>{theme}</h2>
              <ul
                className={`${fr.cx(
                  "fr-grid-row",
                  "fr-grid-row--gutters",
                  "fr-grid-row--left"
                )}`}
              >
                {modeles.map((model) => (
                  <li
                    key={model.slug}
                    className={`${fr.cx(
                      "fr-col-12",
                      "fr-col-sm-12",
                      "fr-col-md-6",
                      "fr-col-lg-6"
                    )} ${li}`}
                  >
                    <Card
                      border
                      desc={summarize(model.description)}
                      horizontal
                      linkProps={{
                        href: `/${getRouteBySource(SOURCES.LETTERS)}/${model.slug}`,
                      }}
                      size="medium"
                      title={model.title}
                      titleAs="h3"
                      enlargeLink
                      classes={{
                        // Hack: start div is render and take some place
                        start: fr.cx("fr-hidden"),
                      }}
                    />
                  </li>
                ))}
              </ul>
            </>
          );
        })}
    </ContainerList>
  );
};

const li = css({
  listStyle: "none!",
});

const list = css({
  maxWidth: 500,
});
