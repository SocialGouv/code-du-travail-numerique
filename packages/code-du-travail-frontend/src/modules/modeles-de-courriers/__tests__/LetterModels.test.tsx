import { getAllByRole, getByLabelText, render } from "@testing-library/react";
import { LetterModels } from "../LetterModels";
import { ModelDescriptionByRootTheme } from "../queries";
import userEvent from "@testing-library/user-event";

const modeles: ModelDescriptionByRootTheme = [
  {
    theme: "Thème A",
    modeles: [
      {
        title: "Modele AA",
        slug: "modele-aa",
        description: "Description modèle AA",
        breadcrumbs: [
          {
            label: "Theme A",
            slug: "theme-a",
            position: 0,
          },
        ],
      },
      {
        title: "Modele AB",
        slug: "modele-ab",
        description: "Description modèle AB",
        breadcrumbs: [
          {
            label: "Theme A",
            slug: "theme-a",
            position: 0,
          },
        ],
      },
    ],
  },
  {
    theme: "Thème B",
    modeles: [
      {
        title: "Modele BA",
        slug: "modele-ba",
        description: "Description modèle BA",
        breadcrumbs: [
          {
            label: "Theme B",
            slug: "theme-b",
            position: 0,
          },
        ],
      },
    ],
  },
];

describe("<LetterModels />", () => {
  it("affiche la liste des modèles de document", () => {
    const { container, getByLabelText } = render(
      <LetterModels modeles={modeles} />
    );
    const selectThemes = getByLabelText("Sélectionnez un thème");
    expect(selectThemes).toBeInTheDocument();
    const themeOptions = getAllByRole(selectThemes, "option");
    expect(themeOptions).toHaveLength(3);
    expect(themeOptions[0]).toHaveTextContent("Tous les thèmes");
    expect(themeOptions[1]).toHaveTextContent("Thème A");
    expect(themeOptions[2]).toHaveTextContent("Thème B");
    expect(container).toMatchSnapshot();
  });
  it("filtre sur le thème sélectionné", () => {
    const { getAllByRole, getByLabelText } = render(
      <LetterModels modeles={modeles} />
    );

    const themes = getAllByRole("heading", { level: 2 });
    expect(themes).toHaveLength(2);
    expect(themes[0]).toHaveTextContent("Thème A");
    expect(themes[1]).toHaveTextContent("Thème B");
    const documents = getAllByRole("heading", { level: 3 });
    expect(documents).toHaveLength(3);
    expect(documents[0]).toHaveTextContent("Modele AA");
    expect(documents[1]).toHaveTextContent("Modele AB");
    expect(documents[2]).toHaveTextContent("Modele BA");

    const selectThemes = getByLabelText("Sélectionnez un thème");
    userEvent.selectOptions(selectThemes, "Thème B");
    const themesFiltered = getAllByRole("heading", { level: 2 });
    expect(themesFiltered).toHaveLength(1);
    expect(themesFiltered[0]).toHaveTextContent("Thème B");
    const documentsFiltered = getAllByRole("heading", { level: 3 });
    expect(documentsFiltered).toHaveLength(1);
    expect(documentsFiltered[0]).toHaveTextContent("Modele BA");
  });
});
