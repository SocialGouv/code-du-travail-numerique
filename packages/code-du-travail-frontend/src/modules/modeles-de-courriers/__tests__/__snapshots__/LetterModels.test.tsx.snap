// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<LetterModels /> affiche la liste des modèles de document 1`] = `
<div>
  <div
    class="fr-grid-row"
  >
    <nav
      aria-label="vous êtes ici :"
      class="fr-breadcrumb fr-mb-2v"
      id="fr-breadcrumb-«r0»"
      role="navigation"
    >
      <button
        aria-controls="breadcrumb-«r1»"
        aria-expanded="false"
        class="fr-breadcrumb__button"
      >
        Voir le fil d’Ariane
      </button>
      <div
        class="fr-collapse"
        id="breadcrumb-«r1»"
      >
        <ol
          class="fr-breadcrumb__list"
        >
          <li>
            <a
              class="fr-breadcrumb__link"
              href="/"
            >
              Accueil
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="fr-breadcrumb__link"
            >
              Modèles de documents
            </a>
          </li>
        </ol>
      </div>
    </nav>
    <div
      class="fr-col-12 fr-mb-4w fr-mb-md-12w"
    >
      <h1
        class="fr-mt-0 fr-mb-6w"
        id="modeles"
      >
        Modèles de documents
      </h1>
      <p
        class="fr-text--md"
      >
        Téléchargez et personnalisez les modèles de documents et de lettres pour vos démarches en lien avec le droit du travail
      </p>
      <div
        class="fr-select-group fr-mt-6w max-w_500"
        id="fr-select-group-«r2»"
      >
        <label
          class="fr-label"
          for="select-«r3»"
        >
          Sélectionnez un thème
        </label>
        <select
          aria-describedby=""
          class="fr-select"
          id="select-«r3»"
        >
          <option
            value=""
          >
            Tous les thèmes
          </option>
          <option
            value="Thème A"
          >
            Thème A
          </option>
          <option
            value="Thème B"
          >
            Thème B
          </option>
        </select>
        <div
          aria-live="polite"
          class="fr-messages-group"
          id="select-«r3»-messages-group"
        />
      </div>
      <h2
        class="fr-mt-6w fr-h3"
      >
        Thème A
      </h2>
      <ul
        class="fr-grid-row fr-grid-row--gutters fr-grid-row--left"
      >
        <li
          class="fr-col-12 fr-col-sm-12 fr-col-md-6 fr-col-lg-6 li-s_none!"
        >
          <div
            class="fr-card fr-enlarge-link fr-card--horizontal"
            id="fr-card-«r5»"
          >
            <div
              class="fr-card__body"
            >
              <div
                class="fr-card__content"
              >
                <h3
                  class="fr-card__title"
                >
                  <a
                    class=""
                    href="/modeles-de-courriers/modele-aa"
                  >
                    Modele AA
                  </a>
                </h3>
                <p
                  class="fr-card__desc"
                >
                  Description modèle AA
                </p>
                <div
                  class="fr-card__start fr-hidden"
                />
                <div
                  class="fr-card__end"
                />
              </div>
            </div>
          </div>
        </li>
        <li
          class="fr-col-12 fr-col-sm-12 fr-col-md-6 fr-col-lg-6 li-s_none!"
        >
          <div
            class="fr-card fr-enlarge-link fr-card--horizontal"
            id="fr-card-«r6»"
          >
            <div
              class="fr-card__body"
            >
              <div
                class="fr-card__content"
              >
                <h3
                  class="fr-card__title"
                >
                  <a
                    class=""
                    href="/modeles-de-courriers/modele-ab"
                  >
                    Modele AB
                  </a>
                </h3>
                <p
                  class="fr-card__desc"
                >
                  Description modèle AB
                </p>
                <div
                  class="fr-card__start fr-hidden"
                />
                <div
                  class="fr-card__end"
                />
              </div>
            </div>
          </div>
        </li>
      </ul>
      <h2
        class="fr-mt-6w fr-h3"
      >
        Thème B
      </h2>
      <ul
        class="fr-grid-row fr-grid-row--gutters fr-grid-row--left"
      >
        <li
          class="fr-col-12 fr-col-sm-12 fr-col-md-6 fr-col-lg-6 li-s_none!"
        >
          <div
            class="fr-card fr-enlarge-link fr-card--horizontal"
            id="fr-card-«r7»"
          >
            <div
              class="fr-card__body"
            >
              <div
                class="fr-card__content"
              >
                <h3
                  class="fr-card__title"
                >
                  <a
                    class=""
                    href="/modeles-de-courriers/modele-ba"
                  >
                    Modele BA
                  </a>
                </h3>
                <p
                  class="fr-card__desc"
                >
                  Description modèle BA
                </p>
                <div
                  class="fr-card__start fr-hidden"
                />
                <div
                  class="fr-card__end"
                />
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
